app/controllers/datasets.js

```diff
@@ -67,6 +67,29 @@ export default Ember.Controller.extend({
     let download_link_geojson = encodeURIComponent(this.get('download_link_geojson'));
     return `http://oneclick.cartodb.com/?file=${download_link_geojson}&provider=MAPC&logo=http://data.mapc.org/img/mapc-color.png`;
   }),
+  formattedMetadata: Ember.computed('model', function() {
+    let metadata = this.get('model.metadata');
+    if ('definition' in metadata) {
+      return Ember.A(this.transformGeospatialMetadata(metadata));
+    } else {
+      return metadata;
+    }
+  }),
+
+  transformGeospatialMetadata(geospatialMetadata) {
+    let columnMetadata = geospatialMetadata['definition']['DEFeatureClassInfo']['GPFieldInfoExs']['GPFieldInfoEx']
+
+    let reformattedColumnMetadata = columnMetadata.map(object => {
+      var reformattedObject = {};
+      if (object.AliasName) {
+        reformattedObject['name'] = object.Name;
+        reformattedObject['alias'] = object.AliasName;
+      }
+      return reformattedObject;
+    });
+
+    return reformattedColumnMetadata;
```

Comment:
For lines 82-91, this would be a JavaScriptier way of doing it:

```js
return columnMetadata.map(column => ({ name: column.Name, alias: column.AliasName }))
                     .filter(column => column.alias);
```